### PR TITLE
Adds interfaces for working with Magic Firewall API

### PR DESCRIPTION
**Pull request by cehrig**
Opening this draft PR for visibility.

This PR implements interfaces for working with the following Firewall Rulesets Apis
- List Firewall Rulesets (`https://api.cloudflare.com/#rulesets-list-rulesets`)
- Get Firewall Ruleset (`https://api.cloudflare.com/#rulesets-get-a-ruleset`)
- Create Firewall Ruleset (`https://api.cloudflare.com/#rulesets-create-ruleset`)
- Delete Firewall Ruleset (`https://api.cloudflare.com/#rulesets-delete-ruleset`)
- Update Firewall Ruleset (`https://api.cloudflare.com/#rulesets-update-ruleset`)

More info on that can be found on the dev-docs
https://developers.cloudflare.com/magic-transit/magic-firewall

Sample Calls:
- List Magic Firewall Ruleset
```
lists, err := api.ListMagicFirewallRulesets(context.Background())
```
- Get Magic Firewall Ruleset
```
lists, err := api.GetMagicFirewallRuleset(context.Background(), "0371f7473dde4d6e913d6951dfc19611")
```
- Create new Magic Firewall Ruleset
```
rules := []cloudflare.MagicFirewallRulesetRule{
	{
		Action:           cloudflare.MagicFirewallRulesetRuleActionSkip,
		ActionParameters: &cloudflare.MagicFirewallRulesetRuleActionParameters{Ruleset: "current"},
		Expression:       "tcp.dstport in { 32768..65534 }",
		Description:      "Allow TCP Ephemeral Ports",
		Enabled:          true,
	},
	{
		Action:           cloudflare.MagicFirewallRulesetRuleActionSkip,
		ActionParameters: &cloudflare.MagicFirewallRulesetRuleActionParameters{Ruleset: "current"},
		Expression:       "udp.dstport in { 32768..65535 }",
		Description:      "Allow UDP Ephemeral Port range",
		Enabled:          true,
	},
	{
		Action:      cloudflare.MagicFirewallRulesetRuleActionBlock,
		Expression:  "ip.len >= 0",
		Description: "Deny All",
		Enabled:     true,
	},
}

lists, err := api.CreateMagicFirewallRuleset(context.Background(), "Magic Transit Ruleset", "Global mitigations", rules)
```
- Delete Magic Firewall Ruleset
```
err = api.DeleteMagicFirewallRuleset(context.Background(), "0371f7473dde4d6e913d6951dfc19611")
```

ToDo:
- ~~Double check API behavior~~
- ~~Acceptance Tests~~

### CODE REVIEW

**Comment by cehrig** [comment]
Opening this for reviews. @patryk and @jacobbednarz I'd be happy if you can have a look :wave: 
Just a heads up... The Delete endpoint does not implement our standard response blob (yet) and this is not going to be added anytime soon, so we should be fine here.

File: magic_firewall_rulesets.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+type UpdateMagicFirewallRulesetResponse struct {
+	Response
+	Result MagicFirewallRuleset `json:"result"`
+}
+
+// ListMagicFirewallRulesets lists all Rulesets for a given account
+//
+// API reference: https://api.cloudflare.com/#rulesets-list-rulesets
+func (api *API) ListMagicFirewallRulesets(ctx context.Context) ([]MagicFirewallRuleset, error) {
+	uri := fmt.Sprintf("/accounts/%s/rulesets", api.AccountID)
```
Comment (jacobbednarz) [inline magic_firewall_rulesets.go:103]: My only hesitation is the lack of validation around usage of `api.AccountID` in a couple of these functions as they are not a part of the signature or validated not empty before being utilised. Within upstream clients (such as the Terraform provider) we can perform a check to speed up the feedback cycle however I do wonder if we are going to make this difficult to debug if someone forgets it as the URL will end up being `/accounts//rulesets` with a routing error from the API gateway. Thoughts?

**Review by jacobbednarz — APPROVED** [review]
I've added a non-blocking comment around validation of the account ID before utilising it however 👍 otherwise. Nice one!

**Comment by cehrig** [comment]
That's a good point! I have added a check for empty account IDs.

**Review by jacobbednarz — APPROVED** [review]
Awesome stuff! I think we'll definitely sprinkle that function and pattern around the other methods once this lands.

**Review by patryk — APPROVED** [review]

**Pull request merged**